tests/middleware/test_max_tasks.py

```diff
@@ -0,0 +1,57 @@
+import time
+from unittest import mock
+
+import remoulade
+from remoulade import Worker
+from remoulade.__main__ import main
+from remoulade.middleware import MaxTasks
+
+
+def test_max_tasks(stub_broker):
+    worker = Worker(stub_broker, worker_timeout=100, worker_threads=1)
+    worker.start()
+    stub_broker.add_middleware(MaxTasks(max_tasks=20))
+
+    result = 0
+
+    @remoulade.actor
+    def do_work():
+        nonlocal result
+        result += 1
+
+    stub_broker.declare_actor(do_work)
+
+    for _ in range(30):
+        do_work.send()
+
+    time.sleep(0.1)
```

Comment:
Is it necessary to do a sleep, we cannot use the woker special fucntion to wait ?

Reply:
The problem here is that only the WorkerThread is stopped and not the Worker. The loop that stops the Worker when one of its WorkerThread is stopped is inside the cli and therefore not executed. As a result, worker.join will never return and cannot be used here. Neither can broker.join as some messages will be left waiting.